### PR TITLE
Docs: Add keycloak oauth setup guide

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

- Add keycloak oauth setup guide
- Update local devenv settings to support teamsync
- Update keycloak instance

image

new devenv keycloak interface

image

### CODE REVIEW

**Review by IevaVasiljeva — APPROVED** [review]
Looks good overall! I've added a few comments and suggestions.

I think we should also add disclaimers to bits that are Enterprise & Cloud Pro only features (ie, team sync).

File: docs/sources/setup-grafana/configure-security/configure-authentication/keycloak/index.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  - configuration
+  - documentation
+  - oauth
+title: Configure Keycloak OAuth2 authentication
+weight: 200
+---
+
+# Configure Keycloak OAuth2 authentication
+
+Refer to [Generic OAuth authentication](../generic-oauth) for extra configuration options available for this provider.
```
Comment (IevaVasiljeva) [inline docs/sources/setup-grafana/configure-security/configure-authentication/keycloak/index.md:18]: Another thought - would it be worth having a little blurb about what this article is about ("Keycloak OAuth2 authentication allows you to ..."), and then having a separate subsection for `##Configure Keycloak OAuth2 in Grafana` or something? Basically, similar to the format that other authN setup guides follow.

File: devenv/docker/blocks/auth/oauth/readme.md [hunk 2]
```diff
@@ -19,13 +19,12 @@ name = Keycloak-OAuth
...
 client_secret = d17b9ea9-bcb1-43d2-b132-d339e55872a8
-empty_scopes = true
+scopes = openid email profile offline_access roles
 email_attribute_path = email
-login_attribute_path = login
-name_attribute_path = name
+login_attribute_path = username
+name_attribute_path = full_name
 auth_url = http://localhost:8087/auth/realms/grafana/protocol/openid-connect/auth
 token_url = http://localhost:8087/auth/realms/grafana/protocol/openid-connect/token
```
Comment (mgyongyosi) [inline devenv/docker/blocks/auth/oauth/readme.md:27]: The endpoints have been changed by the latest version:
```suggestion
auth_url = http://localhost:8087/realms/grafana/protocol/openid-connect/auth
token_url = http://localhost:8087/realms/grafana/protocol/openid-connect/token
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/keycloak/index.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+```
+email
+offline_access
+profile
+roles
+```
+
+> **Warning**: these scopes do not add group claims to the id_token. Without group claims, teamsync will not work. Teamsync is covered further down in this document.
+
+3. For role mapping to work with the example configuration above,
```
[the remark this answers is not in the document]
Comment (Jguer) [reply docs/sources/setup-grafana/configure-security/configure-authentication/keycloak/index.md:79]: :+1:

**Pull request merged**